main.ino

```diff
@@ -70,20 +70,29 @@ void loop()
         Serial.println();
     }
     
-    statusCheckCount = (statusCheckCount + 1) % statusCheckInterval;
-    if (statusCheckCount == 0) {
-        statusText = "";
+    unsigned long timeSinceStart = millis();
+    if (!lastStatusCheckTime
+        || timeSinceStart < lastStatusCheckTime // millis() overflow
+        || timeSinceStart - lastStatusCheckTime >= statusCheckIntervalMs) {
+
+        String statusText = "";
+        lastStatusCheckTime = timeSinceStart;
 
         if (!LBattery.isCharging() && LBattery.level() < 30) {
             statusText += "Battery level is less than 30%! "
         }
         
         if (statusText.length() > 0) {
-            LSMS.beginSMS(statusReportPhone);
-            LSMS.print(statusText);
-            LSMS.endSMS();
+            sendSMS(statusText, statusReportPhone);
         }
     }
   
     delay(500);
 }
+
+void sendSMS(String message, String phoneNumber)
```

Comment:
Number should come first, Michael!
When you start SMS its natural to enter a number or choose contact.

Reply:
No, it shouldn't :) Function's first argument should be function's "subject", all the rest are parameters. In this case, sendSMS's subject is the message, phone is parameter.